src/Authentication/JWT/JWSAdapterInterface.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CodeIgniter\Shield\Authentication\JWT;
+
+use stdClass;
+
+interface JWSAdapterInterface
+{
+    /**
+     * Issues Signed JWT (JWS)
+     *
+     * @param array<mixed>               $payload The payload.
+     * @param string                     $keyset  The key group.
+     *                                            The array key of Config\AuthJWT::$keys.
+     * @param array<string, string>|null $headers An array with header elements to attach.
+     *
+     * @return string JWT (JWS)
+     */
+    public static function encode(
```

Comment:
I'm curious your decision to make the interface and adapters use static methods. This seems like a missed opportunity for injectable dependencies and maybe even a service.

Reply:
Maybe because the Firebase methods were static and the adapter did not have state, I made the interface static.
Removing static is easy. However, I am not sure if it really should be removed.

Reply:
There is no need to be `static`, so I removed.
It seems I don't like static.